### PR TITLE
Enable configuration of ignored tables

### PR DESCRIPTION
**Pull request by gpang**
This allows ignoring a certain set of tables when attaching a database.

### CODE REVIEW

**Comment by gpang** [comment]
this depends on Alluxio/alluxio#10897

File: table/server/master/src/main/java/alluxio/master/table/Database.java [hunk 1]
```diff
@@ -58,20 +60,23 @@
...
+      CatalogConfiguration config) {
     mContext = context;
     mType = type;
     mName = name;
     mTables = new ConcurrentHashMap<>();
     mUdb = udb;
-    mConfig = configMap;
+    mConfig = config;
+    mIgnoreTables = Sets.newHashSet(Splitter.on(",").trimResults().omitEmptyStrings()
+        .split(mConfig.get(CatalogProperty.DB_IGNORE_TABLES)));
```
Comment (ZacBlanco) [inline table/server/master/src/main/java/alluxio/master/table/Database.java]: This looks like the same implementation from `InstancedConfiguration`, but maybe we can extract this into a function in `FormatUtils` or `CommonUtils` so we don't have the code in both places?

Comment (gpang) [reply table/server/master/src/main/java/alluxio/master/table/Database.java]: I extract that part.

File: core/common/src/test/java/alluxio/util/ConfigurationUtilsTest.java [hunk 2]
```diff
@@ -142,6 +143,15 @@ public void testMasterNotConfiguredMessage() {
...
   }
 
+  @Test
+  public void parseAsList() {
+    assertEquals(Lists.newArrayList("a"), ConfigurationUtils.parseAsList("a", ","));
+    assertEquals(Lists.newArrayList("a", "b", "c"), ConfigurationUtils.parseAsList("a,b,c", ","));
+    assertEquals(Lists.newArrayList("a", "b", "c"),
+        ConfigurationUtils.parseAsList(" a , b , c ", ","));
+    assertEquals(Lists.newArrayList("a,b,c"), ConfigurationUtils.parseAsList("a,b,c", ";"));
+  }
```
Comment (ZacBlanco) [inline core/common/src/test/java/alluxio/util/ConfigurationUtilsTest.java]: Can we add a case for `a,,b` to make sure empty strings are omitted?

File: core/common/src/main/java/alluxio/util/ConfigurationUtils.java [hunk 3]
```diff
@@ -358,10 +357,8 @@ public static boolean isHaMode(AlluxioConfiguration conf) {
 
     List<ConfigProperty> configs = new ArrayList<>();
     List<PropertyKey> selectedKeys =
-        conf.keySet().stream()
-            .filter(key -> GrpcUtils.contains(key.getScope(), scope))
-            .filter(key -> key.isValid(key.getName()))
-            .collect(toList());
+        conf.keySet().stream().filter(key -> GrpcUtils.contains(key.getScope(), scope))
+            .filter(key -> key.isValid(key.getName())).collect(toList());
```
Comment (ZacBlanco) [inline core/common/src/main/java/alluxio/util/ConfigurationUtils.java]: Not sure if you updated all this by hand or automatically formatted, but I think the old version of this is easier to read

File: core/common/src/test/java/alluxio/util/ConfigurationUtilsTest.java (hunk 2 above)
Comment (gpang) [reply core/common/src/test/java/alluxio/util/ConfigurationUtilsTest.java]: Good call. Added.

File: core/common/src/main/java/alluxio/util/ConfigurationUtils.java (hunk 3 above)
Comment (gpang) [reply core/common/src/main/java/alluxio/util/ConfigurationUtils.java]: It was auto-formatted. I reverted back.

**Comment by gpang** [comment]
I am updating this with tests and improvements

**Comment by gpang** [comment]
@yuzhu please take a look

File: table/server/master/src/main/java/alluxio/master/table/CatalogProperty.java [hunk 4]
```diff
@@ -26,6 +26,7 @@ private CatalogProperty(String name, String description, String defaultValue) {
     super(name, description, defaultValue);
   }
 
-  public static final CatalogProperty DB_TYPE =
```
Comment (yuzhu) [inline table/server/master/src/main/java/alluxio/master/table/CatalogProperty.java]: why delete this? do we not need it for `GLUE`?

File: table/server/master/src/main/java/alluxio/master/table/CatalogProperty.java [hunk 5]
```diff
@@ -26,6 +26,7 @@ private CatalogProperty(String name, String description, String defaultValue) {
...
-      new CatalogProperty("db.type", "The type of database. Possible values are: hive", "");
+  public static final CatalogProperty DB_IGNORE_TABLES =
+      new CatalogProperty("catalog.db.ignore.tables",
+          "The comma-separated list of table names to ignore from the UDB.", "");
```
Comment (yuzhu) [inline table/server/master/src/main/java/alluxio/master/table/CatalogProperty.java]: does this option affect deletion? should we have a comment about that?

Comment (gpang) [reply table/server/master/src/main/java/alluxio/master/table/CatalogProperty.java]: For now, it is only related to getting tables from udb via udb sync.

**Comment by gpang** [comment]
alluxio-bot, merge this please

**Pull request merged**